### PR TITLE
Don't use the corooteen library anymore

### PR DESCRIPTION
**Pull request by tomaka**
https://github.com/bytecodealliance/wasmtime/pull/2434 has now been merged. This PR updates the code for this new feature and removes usage of `corooteen`.

Fix https://github.com/paritytech/smoldot/issues/276

One of the objectives would ultimately be to publish `smoldot` on crates.io, so depending on a `git` version is kind of "meh".
However, in terms of dependencies, we're trading one crates.io dependency and one git dependency for one git dependency, which makes this trade neutral.

### CODE REVIEW

**Comment by tomaka** [comment]
The way it works is by having a `Shared` struct that is shared by `Rc` between the external functions that `wasmtime` calls and the `Jit` object.

When `wasmtime` calls an external function, we store the information about the call in this `Shared` and return `Pending`.
`wasmtime` in turns propagates this pending to the `Jit`, which reads the content of `Shared`.
Later, we put back the return value in `Shared`, and resume execution. This time, the inner function returns `Ready`.

While this is a bit hacky, it is already how `corooteen` was implemented.

**Comment by tomaka** [comment]
cc @pepyakin (might be interesting)

**Comment by tomaka** [comment]
It seems to sometimes freeze, which is worrisome.

File: src/executor/vm/jit.rs [hunk 1]
```diff
@@ -192,132 +233,60 @@ impl JitPrototype {
...
     pub fn global_value(&mut self, name: &str) -> Result<u32, GlobalValueErr> {
-        match self
-            .coroutine
-            .run(Some(ToCoroutine::GetGlobal(name.to_owned())))
-        {
-            corooteen::RunOut::Interrupted(FromCoroutine::GetGlobalResponse(outcome)) => outcome,
-            _ => unreachable!(),
+        match self.instance.get_export(name) {
+            Some(wasmtime::Extern::Global(g)) => match g.get() {
+                wasmtime::Val::I32(v) => Ok(u32::from_ne_bytes(v.to_ne_bytes())),
```
[the remark this answers is not in the document]
Comment (tomaka) [reply src/executor/vm/jit.rs:248]: It's the only way I know to turn a `i32` into a `u32` with the same bits representation :shrug:

[the remark this answers is not in the document]
Comment (pepyakin) [reply src/executor/vm/jit.rs:248]: But isn't this, as well as a regular cast via `as`, is just a noop? Endianness doesn't matter when you speak about primitive values like here and `as` would just reinterpret the same bits as an unsigned value.

[the remark this answers is not in the document]
Comment (tomaka) [reply src/executor/vm/jit.rs:248]: Yeah, it's a noop. The point is to tell the Rust compiler that we want the `u32` with the same bits as this `i32`.

**Comment by tomaka** [comment]
`wasmtime` 0.24 released. No longer need to depend on git.

**Comment by tomaka** [comment]
> It seems to sometimes freeze, which is worrisome.

I think this was caused by https://github.com/paritytech/smoldot/issues/615, and not by `wasmtime`.
Can't reproduce any freeze after https://github.com/paritytech/smoldot/pull/621

**Comment by tomaka** [comment]
Manage to reproduce a couple of freezes, but they were always networking-related.

**Pull request merged**